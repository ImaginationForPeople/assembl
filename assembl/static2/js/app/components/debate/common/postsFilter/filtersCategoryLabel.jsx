// @flow
import * as React from 'react';
import { Translate } from 'react-redux-i18n';

type Props = {
  labelMsgId: string
};

const FiltersCategoryLabel = ({ labelMsgId }: Props) => {
  const categoryLabelStyle = {
    fontWeight: 'bold',
    textAlign: 'left',
    fontSize: '14px',
    padding: '20px 0 5px 20px',
    color: '#777'
  };
  return (
    <div style={categoryLabelStyle}>
      <strong>
        <Translate value={labelMsgId} />
      </strong>
    </div>
  );
};

export default FiltersCategoryLabel;