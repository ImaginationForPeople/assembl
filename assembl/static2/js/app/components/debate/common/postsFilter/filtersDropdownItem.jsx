// @flow

import * as React from 'react';
import { Translate } from 'react-redux-i18n';

type FiltersDropdownItemProps = {
  inputName: string,
  inputType: 'radio' | 'checkbox',
  item: PostsFilterItem, // FIXME: why PostsFilterMenuItem does not work ?
  onSelectItem: (PostsFilterItem, selected?: boolean) => void,
  selected: boolean
};

class FiltersDropdownItem extends React.Component<FiltersDropdownItemProps> {
  handleClick = (event: any) => {
    this.props.onSelectItem(this.props.item, !this.props.selected);
    event.preventDefault();
    event.stopPropagation();
  };

  render() {
    const { inputName, inputType, item, selected } = this.props;

    const FiltersItemStyle = {
      textAlign: 'left',
      padding: '0 20px',
      fontSize: '14px'
    };

    const FilterItemsLabelStyle = {
      paddingLeft: '5px'
    };

    return (
      <div id={`postsFilterItem-${item.id}`} onClick={this.handleClick} style={FiltersItemStyle}>
        {selected}
        <input id={`postsFilterItem-${item.id}-input`} type={inputType} name={inputName} checked={selected} />
        <label htmlFor={`postsFilterItem-${item.id}-input`} style={FilterItemsLabelStyle}>
          <Translate value={item.labelMsgId} />
        </label>
      </div>
    );
  }
}

export default FiltersDropdownItem;