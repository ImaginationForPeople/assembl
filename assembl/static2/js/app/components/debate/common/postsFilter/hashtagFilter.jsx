// @flow
import * as React from 'react';
import Select from 'react-select';
import { I18n } from 'react-redux-i18n';

type Props = {
  hashtags?: string[],
  selectedHashtag: string,
  setPostsFilterHashtags: (hashtags: string[]) => void,
}

export class DumbHashtagsFilter extends React.Component<Props> {

  selectHashtagValue(hashtag: string | null) {
    if (!hashtag) {
      this.props.setPostsFilterHashtags([]);
    } else {
      this.props.setPostsFilterHashtags([hashtag]);
    }
  }

  render() {
    const { hashtags, selectedHashtag } = this.props;

    const options = (hashtags || []).map(hashtag => ({ label: `#${hashtag}`, value: hashtag }));
    const value = selectedHashtag ? { label: `#${selectedHashtag}`, value: selectedHashtag } : null;

    return (
      <Select
        options={options}
        isSearchable
        placeholder={I18n.t('debate.thread.hashtagsFilter.noOption')}
        closeMenuOnSelect
        isClearable
        value={value}
        onChange={selectedOption => this.selectHashtagValue(selectedOption ? selectedOption.value : null)}
      />
    );
  }
}


export default DumbHashtagsFilter;