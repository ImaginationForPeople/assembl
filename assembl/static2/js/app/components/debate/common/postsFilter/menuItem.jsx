// @flow

import * as React from 'react';
import { MenuItem } from 'react-bootstrap';
import { Translate } from 'react-redux-i18n';

type PostsFilterMenuItemProps = {
  inputName: string,
  inputType: 'radio' | 'checkbox',
  item: PostsFilterItem, // FIXME: why PostsFilterMenuItem does not work ?
  onSelectItem: (PostsFilterItem, selected?: boolean) => void,
  selected: boolean
};

class DumbPostsFilterMenuItem extends React.Component<PostsFilterMenuItemProps> {
  handleClick = (event: any) => {
    event.preventDefault();
    event.stopPropagation();
    this.props.onSelectItem(this.props.item, !this.props.selected);
  };

  render() {
    const { inputName, inputType, item, selected } = this.props;
    return (
      <MenuItem id={`postsFilterItem-${item.id}`} onClick={this.handleClick}>
        <input id={`postsFilterItem-${item.id}-input`} type={inputType} name={inputName} checked={selected} />&nbsp;
        <label htmlFor={`postsFilterItem-${item.id}-input`}>
          <Translate value={item.labelMsgId} />
        </label>
      </MenuItem>
    );
  }
}

export default DumbPostsFilterMenuItem;