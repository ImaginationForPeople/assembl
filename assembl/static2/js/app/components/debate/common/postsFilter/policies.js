// @flow
import { DeletedPublicationStates } from '../../../../constants';

const deletedPublicationStates = Object.keys(DeletedPublicationStates);

/*
 * From a post, get the latest creationDate of live descendants and self
 */
const getLatest: (post: PostWithChildren) => string | null = (post) => {
  let maxDate = post.creationDate;
  if (post.children.length === 0) {
    if (deletedPublicationStates.indexOf(post.publicationState) > -1) {
      return null;
    }
    return maxDate;
  }
  post.children.forEach((p) => {
    const date = getLatest(p);
    if (date && date > maxDate) {
      maxDate = date;
    }
  });
  return maxDate;
};

type PostsComparator = (a: PostWithChildren, b: PostWithChildren) => 1 | 0 | -1; // how to compare groups of posts

const compareOnTopPostCreationDate: PostsComparator = (a: PostWithChildren, b: PostWithChildren) => {
  if (a.creationDate > b.creationDate) {
    return 1;
  } else if (a.creationDate === b.creationDate) {
    return 0;
  }
  return -1;
};

const compareOnLatestCreationDate: PostsComparator = (a: PostWithChildren, b: PostWithChildren) => {
  const firstDate = getLatest(a) || a.creationDate;
  const secondDate = getLatest(b) || b.creationDate;
  if (firstDate > secondDate) {
    return 1;
  } else if (firstDate === secondDate) {
    return 0;
  }
  return -1;
};

const childrenCount = (post: PostWithChildren): number => {
  const children = post.children || [];
  return children.length + children.reduce((acc, child) => acc + (child.children ? 0 : childrenCount(child.children)), 0);
};

const compareOnPopularity: PostsComparator = (a: PostWithChildren, b: PostWithChildren) => {
  const popularityA = a.sentimentCounts.like - a.sentimentCounts.disagree + childrenCount(a);
  const popularityB = b.sentimentCounts.like - b.sentimentCounts.disagree + childrenCount(b);
  if (popularityA === popularityB) {
    return 0;
  } else if (popularityA > popularityB) {
    return 1;
  }
  return -1;
};

export const reverseChronologicalTopPolicy: PostsOrderPolicy = {
  // Recently started threads
  id: 'reverseChronologicalTop',
  postsGroupPolicy: {
    comparator: compareOnTopPostCreationDate,
    reverse: true
  },
  graphqlPostsOrder: 'reverse_chronological',
  labelMsgId: 'debate.thread.postsOrder.reverseChronologicalTop'
};

export const reverseChronologicalLastPolicy: PostsOrderPolicy = {
  // Recently active threads
  id: 'reverseChronologicalLast',
  postsGroupPolicy: {
    comparator: compareOnLatestCreationDate,
    reverse: true
  },
  reversePosts: true,
  graphqlPostsOrder: 'reverse_chronological',
  labelMsgId: 'debate.thread.postsOrder.reverseChronologicalLast'
};

export const chronologicalTopPolicy: PostsOrderPolicy = {
  // Chronological threads
  id: 'chronologicalTop',
  postsGroupPolicy: {
    comparator: compareOnTopPostCreationDate,
    reverse: false
  },
  graphqlPostsOrder: 'chronological',
  labelMsgId: 'debate.thread.postsOrder.chronologicalTop'
};

export const reverseChronologicalFlatPolicy: PostsOrderPolicy = {
  // Newest messages first
  id: 'reverseChronologicalFlat',
  postsGroupPolicy: null,
  graphqlPostsOrder: 'reverse_chronological',
  labelMsgId: 'debate.thread.postsOrder.reverseChronologicalFlat'
};

export const popularityPolicy: PostsOrderPolicy = {
  // Popular messages first
  id: 'popularityFlat',
  postsGroupPolicy: {
    comparator: compareOnPopularity,
    reverse: true
  },
  graphqlPostsOrder: 'popularity',
  labelMsgId: 'debate.thread.postsOrder.popularityFlat'
};

export const fullDisplayPolicy: PostsDisplayPolicy = {
  labelMsgId: 'debate.thread.postsDisplay.full',
  id: 'display-full',
  displayMode: 'full'
};

export const summaryDisplayPolicy: PostsDisplayPolicy = {
  labelMsgId: 'debate.thread.postsDisplay.summary',
  id: 'display-summary',
  displayMode: 'summary',
  type: 'choice'
};

export const onlyMyPostFilterPolicy: PostsFilterPolicy = {
  anonymous: false,
  excludedPolicies: ['myPostsAndAnswers'],
  filterField: 'onlyMyPosts',
  id: 'filter-onlyMyPosts',
  labelMsgId: 'debate.thread.postsFilters.onlyMyPosts',
  type: 'choice'
};

export const myPostsAndAnswersFilterPolicy: PostsFilterPolicy = {
  anonymous: false,
  excludedPolicies: ['onlyMyPosts'],
  filterField: 'myPostsAndAnswers',
  id: 'filter-myPostsAndAnswers',
  labelMsgId: 'debate.thread.postsFilters.myPostsAndAnswers',
  type: 'choice'
};

export const hashtagsFilterPolicy: PostsFilterPolicy = {
  anonymous: true,
  excludedPolicies: [],
  filterField: 'hashtags',
  labelMsgId: 'debate.thread.postsFilters.hashtags',
  id: 'filter-hashtags',
  type: 'select'
};

export const defaultOrderPolicy = reverseChronologicalLastPolicy;
export const defaultDisplayPolicy = fullDisplayPolicy;
export const defaultPostsFiltersStatus: PostsFiltersStatus = {
  myPostsAndAnswers: false,
  onlyMyPosts: false,
  hashtags: []
};