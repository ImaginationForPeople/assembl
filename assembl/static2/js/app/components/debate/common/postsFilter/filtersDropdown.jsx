// @flow
import * as React from 'react';
import { DropdownButton } from 'react-bootstrap';
import debounce from 'lodash/debounce';
import classNames from 'classnames';
import { I18n } from 'react-redux-i18n';

import FiltersDropdownItem from './filtersDropdownItem';
import PostsFilterButton from './button';
import PostsFilterButtons from './buttons';
import PostsFiltersCategoryLabel from './filtersCategoryLabel';
import { getConnectedUserId } from '../../../../utils/globalFunctions';
import { withScreenHeight } from '../../../common/screenDimensions';
import { defaultDisplayPolicy, defaultOrderPolicy, defaultPostsFiltersStatus } from './policies';
import { ICO_FILTER } from '../../../../constants';

type Props = {
  defaultDisplayPolicy: PostsDisplayPolicy,
  defaultOrderPolicy: PostsOrderPolicy,
  defaultPostsFiltersStatus: PostsFiltersStatus,
  postsDisplayPolicies: PostsDisplayPolicy[],
  postsFiltersPolicies: PostsFilterPolicy[],
  postsOrderPolicies: PostsOrderPolicy[],
  postsDisplayPolicy: PostsDisplayPolicy,
  postsFiltersStatus: PostsFiltersStatus,
  postsOrderPolicy: PostsOrderPolicy,
  stickyOffset: number,
  stickyTopPosition: number,
  screenHeight: number,
  setPostsFilterPolicies: (
    postsDisplay: PostsDisplayPolicy,
    postsOrder: PostsOrderPolicy,
    postsFiltersStatus: PostsFiltersStatus
  ) => void
};

type State = {
  open: boolean,
  selectedPostsDisplayPolicy: PostsDisplayPolicy,
  selectedPostsFiltersStatus: PostsFiltersStatus,
  selectedPostsOrderPolicy: PostsOrderPolicy,
  sticky: boolean
};

/*
 * Generic post filter component. Specific filter components are in subfolders survey/, thread/, etc.
 * */
export class FiltersDropdown extends React.Component<Props, State> {
  static defaultProps = {
    defaultOrderPolicy: defaultOrderPolicy,
    defaultDisplayPolicy: defaultDisplayPolicy,
    defaultPostsFiltersStatus: defaultPostsFiltersStatus
  };

  componentWillMount(): void {
    const { postsDisplayPolicy, postsOrderPolicy, postsFiltersStatus } = this.props;
    this.setState({
      open: false,
      sticky: true,
      selectedPostsFiltersStatus: { ...postsFiltersStatus },
      selectedPostsDisplayPolicy: postsDisplayPolicy,
      selectedPostsOrderPolicy: postsOrderPolicy
    });
  }

  componentDidMount() {
    window.addEventListener('scroll', this.setButtonPosition);
  }

  componentWillUnmount = () => {
    window.removeEventListener('scroll', this.setButtonPosition);
  };

  onToggle = () => {
    const { open } = this.state;
    this.setState({ open: !open });
  };

  setButtonPosition = debounce(() => {
    const { screenHeight, stickyOffset } = this.props;
    if (window.pageYOffset > screenHeight - stickyOffset) {
      // Show the button when we scrolled minimum the height of the window.
      this.setState({ sticky: true });
    } else {
      this.setState({ sticky: false });
    }
  }, 100);

  selectPostsDisplayPolicy = (postsDisplayPolicy: any) => {
    // FIXME use PostsDisplayPolicy type
    this.setState({ selectedPostsDisplayPolicy: postsDisplayPolicy });
  };

  selectPostsOrderPolicy = (postsOrderPolicy: any) => {
    // FIXME use PostsOrderPolicy type
    this.setState({ selectedPostsOrderPolicy: postsOrderPolicy });
  };

  selectPostsFilter = (postsFilterPolicy: any, selected?: boolean) => {
    // FIXME use PostsFilterPolicy type
    const selectedPostsFiltersStatus = { ...this.state.selectedPostsFiltersStatus };

    selectedPostsFiltersStatus[postsFilterPolicy.filterField] = !!selected;

    if (selected) {
      // unselect excluded policies
      postsFilterPolicy.excludedPolicies.forEach((policyId: string) => {
        selectedPostsFiltersStatus[policyId] = false;
      });
    }
    this.setState({ selectedPostsFiltersStatus: selectedPostsFiltersStatus });
  };

  resetPolicies = () => {
    this.selectPostsDisplayPolicy(defaultDisplayPolicy);
    this.selectPostsOrderPolicy(defaultOrderPolicy);
    this.setState({ selectedPostsFiltersStatus: this.props.defaultPostsFiltersStatus });
    this.props.setPostsFilterPolicies(
      this.props.defaultDisplayPolicy,
      this.props.defaultOrderPolicy,
      this.props.defaultPostsFiltersStatus
    );
    this.setState({ open: false });
  };

  savePolicies = () => {
    this.props.setPostsFilterPolicies(
      this.state.selectedPostsDisplayPolicy,
      this.state.selectedPostsOrderPolicy,
      this.state.selectedPostsFiltersStatus
    );
    this.setState({ open: false });
  };

  render() {
    const { selectedPostsDisplayPolicy, selectedPostsFiltersStatus, selectedPostsOrderPolicy, sticky } = this.state;
    const { postsDisplayPolicies, postsOrderPolicies, postsFiltersPolicies, stickyTopPosition } = this.props;
    const userIsConnected: boolean = !!getConnectedUserId();
    return (
      <div
        className={classNames(['posts-filter-button', sticky ? 'sticky' : null])}
        style={{ top: sticky ? stickyTopPosition : null }}
      >
        <DropdownButton
          pullRight
          open={this.state.open}
          onToggle={this.onToggle}
          title={
            <img
              height={24}
              width={24}
              src={ICO_FILTER}
              alt={I18n.t('debate.thread.sortFilterPosts')}
              title={I18n.t('debate.thread.sortFilterPosts')}
            />
          }
          id="postsFilter-dropdown"
        >
          <PostsFiltersCategoryLabel labelMsgId="debate.thread.sortPosts" />

          {postsOrderPolicies.map(item => (
            <FiltersDropdownItem
              key={item.id}
              item={item}
              selected={item.id === selectedPostsOrderPolicy.id}
              inputType="radio"
              inputName="postsOrder"
              onSelectItem={this.selectPostsOrderPolicy}
              eventKey={item.id}
            />
          ))}
          <PostsFiltersCategoryLabel labelMsgId="debate.thread.overviewPosts" />

          {postsDisplayPolicies.map(item => (
            <FiltersDropdownItem
              key={item.id}
              item={item}
              selected={item.id === selectedPostsDisplayPolicy.id}
              inputType="radio"
              inputName="postsDisplay"
              onSelectItem={this.selectPostsDisplayPolicy}
              eventKey={item.id}
            />
          ))}

          {userIsConnected && (
            <React.Fragment>
              <PostsFiltersCategoryLabel labelMsgId="debate.thread.filterPosts" />
              {postsFiltersPolicies.map(item => (
                <FiltersDropdownItem
                  key={item.id}
                  item={item}
                  selected={selectedPostsFiltersStatus[item.filterField]}
                  inputType="checkbox"
                  inputName={item.filterField}
                  onSelectItem={this.selectPostsFilter}
                  eventKey={item.id}
                />
              ))}
            </React.Fragment>
          )}
          <PostsFilterButtons>
            <PostsFilterButton
              id="postsFilter-button-reset"
              onClick={this.resetPolicies}
              className="button-cancel button-dark"
              i18nTitle="debate.thread.postsOrder.cleanFilter"
              bsStyle="default"
            />
            <PostsFilterButton
              id="postsFilter-button-save"
              style={{ textAlign: 'right', float: 'right' }}
              className="button-submit button-dark"
              onClick={this.savePolicies}
              i18nTitle="ok"
              bsStyle="success"
            />
          </PostsFilterButtons>
        </DropdownButton>
      </div>
    );
  }
}

export default withScreenHeight(FiltersDropdown);