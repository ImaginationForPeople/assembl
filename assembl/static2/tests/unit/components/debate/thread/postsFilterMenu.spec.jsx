import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { FiltersDropdown } from '../../../../../js/app/components/debate/common/postsFilter/filtersDropdown';
import PostsFilterMenu from '../../../../../js/app/components/debate/common/postsFilter/thread/menu';
import {
  defaultDisplayPolicy,
  defaultOrderPolicy,
  defaultPostsFiltersStatus,
  reverseChronologicalTopPolicy,
  summaryDisplayPolicy
} from '../../../../../js/app/components/debate/common/postsFilter/policies';
import {
  postsDisplayPolicies,
  postsFiltersPolicies,
  postsOrderPolicies
} from '../../../../../js/app/components/debate/common/postsFilter/thread/policies';

configure({ adapter: new Adapter() });

jest.mock('../../../../../js/app/utils/globalFunctions', () => ({
  getConnectedUserId: jest.fn(() => '1234567890')
}));

describe('PostsFilterMenu component', () => {
  const mockStore = configureStore();
  const initialState = {
    threadFilter: {
      postsOrderPolicy: defaultOrderPolicy,
      postsFiltersStatus: defaultPostsFiltersStatus,
      postsDisplayPolicy: defaultDisplayPolicy
    }
  };
  const props = {};
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should render a filter with sort by recently started threads and full display selected', () => {
    const component = renderer.create(<PostsFilterMenu store={store} {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should rollback when reset is clicked', () => {
    const wrapper = mount(<PostsFilterMenu {...props} store={store} />);

    const defaultSortItem = () => wrapper.find('div#postsFilterItem-reverseChronologicalLast');
    const nonDefaultSortItem = () => wrapper.find('div#postsFilterItem-reverseChronologicalTop');
    const defaultDisplayItem = () => wrapper.find('div#postsFilterItem-display-full');
    const nonDefaultDisplayItem = () => wrapper.find('div#postsFilterItem-display-summary');
    const onlyMyPostsItem = () => wrapper.find('div#postsFilterItem-filter-onlyMyPosts');

    const resetButton = () => wrapper.find('button#postsFilter-button-reset');

    nonDefaultSortItem().simulate('click');
    nonDefaultDisplayItem().simulate('click');
    expect(
      nonDefaultSortItem()
        .find('input')
        .props().checked
    ).toBe(true);
    expect(
      nonDefaultDisplayItem()
        .find('input')
        .props().checked
    ).toBe(true);

    resetButton().simulate('click');
    expect(
      nonDefaultSortItem()
        .find('input')
        .props().checked
    ).toBe(false);
    expect(
      nonDefaultDisplayItem()
        .find('input')
        .props().checked
    ).toBe(false);
    expect(
      defaultSortItem()
        .find('input')
        .props().checked
    ).toBe(true);
    expect(
      defaultDisplayItem()
        .find('input')
        .props().checked
    ).toBe(true);
    expect(
      onlyMyPostsItem()
        .find('input')
        .props().checked
    ).toBe(false);
  });

  it('should be saved when saved is clicked', () => {
    const setPostsFilterPolicies = jest.fn();
    const component = (
      <FiltersDropdown
        postsDisplayPolicy={defaultDisplayPolicy}
        postsOrderPolicy={defaultOrderPolicy}
        postsFiltersStatus={defaultPostsFiltersStatus}
        setPostsFilterPolicies={setPostsFilterPolicies}
        postsFiltersPolicies={postsFiltersPolicies}
        postsOrderPolicies={postsOrderPolicies}
        postsDisplayPolicies={postsDisplayPolicies}
      />
    );
    const wrapper = mount(component);

    const nonDefaultSortItem = () => wrapper.find('div#postsFilterItem-reverseChronologicalTop');
    const nonDefaultDisplayItem = () => wrapper.find('div#postsFilterItem-display-summary');
    const onlyMyPostsItem = () => wrapper.find('div#postsFilterItem-filter-onlyMyPosts');
    const saveButton = () => wrapper.find('button#postsFilter-button-save');

    nonDefaultSortItem().simulate('click');
    nonDefaultDisplayItem().simulate('click');
    onlyMyPostsItem().simulate('click');
    saveButton().simulate('click');
    expect(setPostsFilterPolicies).toHaveBeenCalledTimes(1);
    expect(setPostsFilterPolicies).toHaveBeenCalledWith(summaryDisplayPolicy, reverseChronologicalTopPolicy, {
      myPostsAndAnswers: false,
      onlyMyPosts: true
    });
  });
});