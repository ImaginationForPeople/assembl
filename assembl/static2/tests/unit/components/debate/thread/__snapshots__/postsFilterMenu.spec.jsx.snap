// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsFilterMenu component should render a filter with sort by recently started threads and full display selected 1`] = `
<div
  className="posts-filter-button sticky"
  style={
    Object {
      "top": undefined,
    }
  }
>
  <div
    className="dropdown btn-group"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-default"
      disabled={false}
      id="postsFilter-dropdown"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      type="button"
    >
      <img
        alt="Sort / Filter"
        height={24}
        src="https://s3-eu-west-1.amazonaws.com/bluenove-assembl-images/filter.svg"
        title="Sort / Filter"
        width={24}
      />
       
      <span
        className="caret"
      />
    </button>
    <ul
      aria-labelledby="postsFilter-dropdown"
      className="dropdown-menu dropdown-menu-right"
      role="menu"
    >
      <li
        className="disabled"
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <strong>
            <span>
              Sort
            </span>
          </strong>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-reverseChronologicalTop"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={false}
            id="postsFilterItem-reverseChronologicalTop-input"
            name="postsOrder"
            onChange={[Function]}
            type="radio"
          />
           
          <label
            htmlFor="postsFilterItem-reverseChronologicalTop-input"
          >
            <span>
              Recently active threads
            </span>
          </label>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-reverseChronologicalLast"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={true}
            id="postsFilterItem-reverseChronologicalLast-input"
            name="postsOrder"
            onChange={[Function]}
            type="radio"
          />
           
          <label
            htmlFor="postsFilterItem-reverseChronologicalLast-input"
          >
            <span>
              Recently started threads
            </span>
          </label>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-chronologicalTop"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={false}
            id="postsFilterItem-chronologicalTop-input"
            name="postsOrder"
            onChange={[Function]}
            type="radio"
          />
           
          <label
            htmlFor="postsFilterItem-chronologicalTop-input"
          >
            <span>
              Chronological order
            </span>
          </label>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-reverseChronologicalFlat"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={false}
            id="postsFilterItem-reverseChronologicalFlat-input"
            name="postsOrder"
            onChange={[Function]}
            type="radio"
          />
           
          <label
            htmlFor="postsFilterItem-reverseChronologicalFlat-input"
          >
            <span>
              Newest messages first
            </span>
          </label>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-popularityFlat"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={false}
            id="postsFilterItem-popularityFlat-input"
            name="postsOrder"
            onChange={[Function]}
            type="radio"
          />
           
          <label
            htmlFor="postsFilterItem-popularityFlat-input"
          >
            <span>
              Most popular messages first
            </span>
          </label>
        </a>
      </li>
      <li
        className="disabled"
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <strong>
            <span>
              Overview
            </span>
          </strong>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-display-full"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={true}
            id="postsFilterItem-display-full-input"
            name="postsDisplay"
            onChange={[Function]}
            type="radio"
          />
           
          <label
            htmlFor="postsFilterItem-display-full-input"
          >
            <span>
              Full messages
            </span>
          </label>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-display-summary"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={false}
            id="postsFilterItem-display-summary-input"
            name="postsDisplay"
            onChange={[Function]}
            type="radio"
          />
           
          <label
            htmlFor="postsFilterItem-display-summary-input"
          >
            <span>
              Message previews
            </span>
          </label>
        </a>
      </li>
      <li
        className="disabled"
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <strong>
            <span>
              Filter
            </span>
          </strong>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-filter-onlyMyPosts"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={false}
            id="postsFilterItem-filter-onlyMyPosts-input"
            name="onlyMyPosts"
            onChange={[Function]}
            type="checkbox"
          />
           
          <label
            htmlFor="postsFilterItem-filter-onlyMyPosts-input"
          >
            <span>
              My posts
            </span>
          </label>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          id="postsFilterItem-filter-myPostsAndAnswers"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <input
            checked={false}
            id="postsFilterItem-filter-myPostsAndAnswers-input"
            name="myPostsAndAnswers"
            onChange={[Function]}
            type="checkbox"
          />
           
          <label
            htmlFor="postsFilterItem-filter-myPostsAndAnswers-input"
          >
            <span>
              Answers to my posts
            </span>
          </label>
        </a>
      </li>
      <li
        className=""
        role="presentation"
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          <button
            className="button-cancel button-dark btn btn-default"
            disabled={false}
            id="postsFilter-button-reset"
            onClick={[Function]}
            type="button"
          >
            <span>
              Clean filters
            </span>
          </button>
          <button
            className="button-submit button-dark btn btn-success"
            disabled={false}
            id="postsFilter-button-save"
            onClick={[Function]}
            style={
              Object {
                "float": "right",
                "textAlign": "right",
              }
            }
            type="button"
          >
            <span>
              Ok
            </span>
          </button>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
